shade-memcached/src/main/scala/shade/memcached/Codec.scala

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2012-2017 by its authors. Some rights reserved.
+ * See the project homepage at: https://github.com/monix/shade
+ *
+ * Licensed under the MIT License (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy
+ * of the License at:
+ *
+ * https://github.com/monix/shade/blob/master/LICENSE.txt
+ */
+
+package shade.memcached
```

Comment:
I don't know how to handle it otherwise. Because we are now using the high-level API, we need to work with `Transcoder`, which isn't a bad thing because we can now also work with `flags`.

Bringing in SpyMemcached's `Transcoder` implementations has these benefits:

1. data persisted by Shade can be read by SpyMemcached - the problem is SpyMemcached `Transcoder` implementations reject values if they aren't stored with proper `flags` - so with the current version you'd have to implement your own `Transcoder` that just deals with `Array[Byte]` like the current `Codec` does
2. the `SerializingTranscoder` in SpyMemcached begins compressing data in case it goes over a certain size, which is cool

I'm not sure if making `Codec` specific to SpyMemcached is bad, because by introducing usage of those `flags`, that's an implementation detail that doesn't seem reusable.